remix/app/routes/_app.newsletter.update.$id.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect } from 'react';
+
+import { ActionArgs, json } from '@remix-run/node';
+import { Form, useActionData, useNavigate } from '@remix-run/react';
+import { withZod } from '@remix-validated-form/with-zod';
+import { validationError } from 'remix-validated-form';
+import { z } from 'zod';
+
+import { addNotification } from '../helpers/localStorage.helper';
+import appHandler from '../lib/handler/app.handler';
+import NewsletterRepository from '../repositories/newsletter.server';
+
+const validator = withZod(
+  z.object({
+    name: z.string().min(1, { message: 'Name is required' }),
+    content: z.string().min(1, { message: 'Content is required' }),
+  })
+);
+
+export async function action({ request, params }: ActionArgs) {
+  return appHandler(request, async (user) => {
+    const result = await validator.validate(await request.formData());
+
+    if (result.error) {
+      return validationError(result.error);
+    }
+
+    const newsletterExists = await NewsletterRepository.findOne({
+      where: { id: params.id },
+    });
+
+    if (!newsletterExists) {
+      return validationError({
+        fieldErrors: {
+          Error: 'Newsletter not found',
+        },
+      });
+    }
+
+    if (newsletterExists.userId !== user.id) {
+      return validationError({
+        fieldErrors: {
+          Error: 'You are not allowed to update this newsletter',
+        },
+      });
+    }
+
+    const newsletterUpdated = await NewsletterRepository.update({
```

Comment:
Or you can use [Prisma's updateMany](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update-multiple-records), which will allow us to do all 3 things(check newsletter id exists, check if the user of the newsletter is the current user, and update the newsletter) at once with only 1 query 🤣 In case you want to give it a try, here is how [I used it](https://github.com/nimblehq/js-framework-benchmark/pull/117/files#diff-e2c82a3d6cd014d8235827328f63df5db8efb48bbd67175aad18498f1950b96dR53-R63).

Reply:
Yep, that's a good option I migrated to that 😁, Thanks for the suggestion.

Improved here: [d426473](https://github.com/nimblehq/js-framework-benchmark/pull/126/commits/d426473d63ea5e96d4c67a790b23169b39f78847)